pkg/plugins/policies/meshhttproute/plugin/v1alpha1/listeners.go

```diff
@@ -148,17 +136,26 @@ func generateListeners(
 	return resources, nil
 }
 
-// prepareRoutes handles the always present, catch all default route
-func prepareRoutes(
-	toRules rules.Rules,
-	svc meshroute_xds.DestinationService,
-) []api.Route {
-	// policy matching for real MeshService is not yet ready
-	conf := rules.ComputeConf[api.PolicyDefault](toRules, core_rules.MeshService(svc.ServiceName))
+func computeConf(toRules rules.ToRules, svc meshroute_xds.DestinationService, meshCtx xds_context.MeshContext) *api.PolicyDefault {
+	// compute for old MeshService
+	conf := rules.ComputeConf[api.PolicyDefault](toRules.Rules, core_rules.MeshService(svc.ServiceName))
 	switch svc.BackendRef.Kind {
 	case common_api.MeshExternalService:
```

Comment:
i feel like this code is being duplicated looking at the MeshTCPRoute PR, can we do this in one place?

Reply:
made public to reuse it for TCPRoute